CHANGELOG.md

```diff
@@ -18,6 +19,47 @@ CHANGELOG
 
 </details>
 
+Swift 4.1
+---------
+
+* [SE-0185][]
+
+  Structs and enums now automatically receive synthesized implementations of `Equatable` and/or `Hashable` if all of their stored members conform to those protocols. You must explicitly declare that the type conforms to `Equatable`/`Hashable` (not in an extension), but the compiler will provide the implementation of `==`/`hashValue` for you if you do not write one yourself.
+
+  For structs, all stored properties must conform to `Equatable`/`Hashable`:
+
+  ```swift
+  public struct Point: Hashable {
+    public let x: Int
+    public let y: Int
+
+    public init(x: Int, y: Int) {
+      self.x = x
+      self.y = y
+    }
+  }
+
+  Point(3, 0) == Point(0, 3)  // false
+  Point(3, 0) == Point(3, 0)  // true
+  Point(3, 0).hashValue       // -2942920663782199421
+  ```
+
+  For enums, each case must either have no associated values or all of its associated values must conform to `Equatable`/`Hashable`:
+
+  ```swift
+  public enum Token: Hashable {
+    case comma
+    case identifier(String)
```

Comment:
You can also have a case with a tuple type, right? Might be worth adding an example of that too.

Reply:
Not yet, unfortunately. The proposal and implementation only covered types where all stored members strictly conform to the protocol. There's no special handling for tuples yet (I'm happy to work on it as a follow-up).

That also means that the somewhat obvious case of `Optional` isn't handled yet, either. My hope was that conditional conformances would be a reality in time for this feature to pick it up, but perhaps that should be added as a special case as well in the meantime?